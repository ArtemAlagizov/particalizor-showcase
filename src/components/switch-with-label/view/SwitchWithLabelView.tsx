import React from "react";
import { Styled } from "../styled";
import { ISwitchWithLabelView } from "./ISwitchWithLabelView";
import { Switch } from "antd";

const SwitchWithLabelView: React.FC<ISwitchWithLabelView> = ({
  onChange,
  defaultValue,
  label
}) => {
  return (
    <Styled.SwitchWithLabelContainer>
      <Styled.SwitchLabel>{label}</Styled.SwitchLabel>
      <Styled.SwitchContainer>
        <Switch defaultChecked={defaultValue} onChange={onChange} />
      </Styled.SwitchContainer>
    </Styled.SwitchWithLabelContainer>
  );
};

export default SwitchWithLabelView;
