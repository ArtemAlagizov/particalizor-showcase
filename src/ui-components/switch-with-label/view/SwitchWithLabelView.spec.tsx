import React from "react";
import { shallow } from "enzyme";
import SwitchWithLabelView from "./SwitchWithLabelView";

describe("Given a SwitchWithLabelView", () => {
  describe("when rendering", () => {
    it("should match snapshot", () => {
      const component = shallow(
        <SwitchWithLabelView
          onChange={jest.fn()}
          label={""}
          defaultValue={false}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
