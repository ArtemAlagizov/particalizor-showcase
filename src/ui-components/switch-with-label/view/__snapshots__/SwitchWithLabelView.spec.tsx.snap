// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a SwitchWithLabelView when rendering should match snapshot 1`] = `
<styled.div>
  <styled.div />
  <styled.div>
    <Switch
      defaultChecked={false}
      onChange={[MockFunction]}
    />
  </styled.div>
</styled.div>
`;
